app/assets/javascripts/initializers/initializeBodyData.js

```diff
@@ -47,6 +47,9 @@ function fetchBaseData() {
             ga('set', 'userId', JSON.parse(json.user).id);
           }
         }, 400);
+      } else {
```

Comment:
Refactor (blocking): I was looking at some code related to this and we'd want to clear local storage too. Adding the removal of `document.body.dataset.user` in there probably makes sense. See https://github.com/forem/forem/blob/438699a342b5c938b0859f3f7cc317e14620a687/app/assets/javascripts/utilities/browserStoreCache.js#L3-L25.	Also, it's only used in a couple of places, so maybe even rework that function.

![image](https://user-images.githubusercontent.com/833231/111479551-7311cf80-8707-11eb-9ef9-e8366cb420d0.png)

Reply:
@nickytonline I added a call to that function, I'm not sure we should add messing with this object to that function since it is pretty explicitly dealing with localStorage only. It could definitely be refactored though.